web-common-marlowe/src/Marlowe/Extended.purs

```diff
@@ -38,16 +38,31 @@ type MetaData
     , choiceDescriptions :: Map String String
     }
 
+emptyContractMetadata :: MetaData
+emptyContractMetadata =
+  { contractType: Other
+  , contractName: ""
+  , contractDescription: ""
+  , roleDescriptions: mempty
+  , slotParameterDescriptions: mempty
+  , valueParameterDescriptions: mempty
+  , choiceDescriptions: mempty
+  }
+
```

Comment:
We could, but I think it would be worse user experience.

The problem most clear to me would be deleting (but I think it applies to adding too). Imagine the following scenario:
- The user writes a contract and fills in the metadata for all the choices in the contract
- Then the user decides to change the name of one of the choices in the contract

If we just update the metadata to match the contract, the information the user filled for the old name of the choice that was renamed just disappears magically. This way the user can see the change required in the metadata, but no information is lost until the button for deleting the old choice is pressed, so the user could copy paste for example.

Also the user may not want to have metadata for some of the things, and I think it is cleaner to not have them in the map than just having an empty string.

Another advantage is that it gives a sense of locality, only changes done in the metadata tab affect the metadata, things done elsewhere don't.

Reply:
What happens at the moment if I have metadata for a choice, and then I delete/change that choiceId? Does the old key stick around in the metadata?

I take your point, but I also think there's a lot to be said for ensuring that metadata keys match up with the template. That's easy if they always depend on the template. Maybe we could show an error/warning on a metadata key that no longer corresponds to anything in the contract? 🤔

Reply:
That is what the current implementation does :)
But for the dashboard I wouldn't bother, just pick the ones that are present and ignore the rest, I think

Reply:
Ah, you've thought of everything... 👍  🏅 

Yeah, the dashboard will just show for each key in the template, and ignore the rest.